tests/integration.rs

```diff
@@ -482,6 +525,99 @@ mod integration_tests {
         }
     }
 
+    /// Controller for running Nomad.
+    pub struct NomadTestController {
+        pub url: String,
+        nomad_handle: Child,
+    }
+
+    impl NomadTestController {
+        pub async fn new() -> Result<NomadTestController> {
+            let url = "127.0.0.1:4646".to_string();
+
+            let mut nomad_handle = Command::new(get_process(NOMAD_BINARY))
+                .args(["agent", "-dev"])
+                .spawn()
+                .with_context(|| "executing nomad")?;
+
+            wait_tcp(&url, &mut nomad_handle, NOMAD_BINARY).await?;
+
+            Ok(Self { url, nomad_handle })
+        }
+    }
+
+    impl Drop for NomadTestController {
+        fn drop(&mut self) {
+            let _ = self.nomad_handle.kill();
+        }
+    }
+
+    /// Controller for running Hippo.
+    pub struct HippoTestController {
+        pub url: String,
+        pub client: hippo::Client,
+        hippo_handle: Child,
+    }
+
+    impl HippoTestController {
+        pub async fn new(bindle_url: &str) -> Result<HippoTestController> {
+            let url = format!("http://127.0.0.1:{}", get_random_port()?);
+
+            let mut hippo_handle = Command::new(get_process(HIPPO_BINARY))
+                .env("ASPNETCORE_URLS", &url)
+                .env("Nomad__Driver", "raw_exec")
+                .env("Nomad__Datacenters__0", "dc1")
+                .env("Database__Driver", "sqlite")
+                .env(
+                    "ConnectionStrings__Database",
+                    "Data Source=hippo.db;Cache=Shared",
```

Comment:
On repeated runs of this integration test, I'm seeing:
```
      Executed endpoint 'Hippo.Web.Api.AccountController.Register (Hippo.Web)'
Error: One or more validation errors occurred. {"UserName": ["The specified username already exists."]}
thread 'integration_tests::test_spin_deploy' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/test/src/lib.rs:186:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test integration_tests::test_spin_deploy ... FAILED

failures:

failures:
    integration_tests::test_spin_deploy

test result: FAILED. 9 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 12.25s
```

Do we need to also be sure to clean up or delete `hippo.db` after (or prior to) each run?

Reply:
Yes, I also wonder about this. After I read https://github.com/fermyon/installer/issues/36#issuecomment-1155234416, I realize that we can use `inmemory` for testing, so we can rerun it.

Reply:
Oh, good point.  Yes, we should try using `inmemory` here.

Reply:
Ah, you already made the change.  Thanks!